ocf_data_sampler/config/model.py

```diff
@@ -40,21 +40,44 @@ class General(Base):
     )
 
 
-class DataSourceMixin(Base):
-    """Mixin class, to add forecast and history minutes"""
+class TimeWindowMixin(Base):
+    """Mixin class, to add interval start, end and resolution minutes"""
 
-    forecast_minutes: int = Field(
+    time_resolution_minutes: int = Field(
         ...,
-        ge=0,
-        description="how many minutes to forecast in the future. ",
+        gt=0,
+        description="The temporal resolution of the data in minutes",
     )
-    history_minutes: int = Field(
+
+    interval_end_minutes: int = Field(
         ...,
-        ge=0,
-        description="how many historic minutes to use. ",
+        description="Data interval ends at `t0 + interval_end_minutes`",
+    )
+    interval_start_minutes: int = Field(
+        ...,
+        description="Data interval starts at `t0 + interval_start_minutes`",
     )
 
 
+    @model_validator(mode='after')
+    def check_interval_range(cls, values):
+        if values.interval_start_minutes > values.interval_end_minutes:
+            raise ValueError('interval_start_minutes must be <= interval_end_minutes')
```

Comment:
Nitpick, but the error message doesn't say what the test doe: one allows for equality, one doesn't